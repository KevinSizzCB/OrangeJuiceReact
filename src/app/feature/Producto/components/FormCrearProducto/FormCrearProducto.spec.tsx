import * as React from 'react';
import { RenderResult, fireEvent, render, wait } from '@testing-library/react';
import { SinonStub, stub } from 'sinon';
import { FormCrearProducto } from './';
import { setTextEvent } from 'app/shared/utils/test';

describe('FormCrearProducto test', () => {
  let componentWrapper: RenderResult;
  let componentProps: React.ComponentProps<typeof FormCrearProducto> & {
    onSubmit: SinonStub;
  };

  beforeEach(() => {
    componentProps = {
      formTitle: 'Form test',
      onSubmit: stub(),
      usuario: { edad: 10, id: 0, nombre: 'Juan' },
    };
    componentWrapper = render(<FormCrearProducto {...componentProps} />);
  });

  it('should match snapshot', () => {
    expect(componentWrapper.container).toMatchSnapshot();
  });

  it('should fail on submit all fields missing', async () => {
    const elem = componentWrapper.container;
    const submitButton = elem.querySelector('button[type="submit"]');

    await wait(() => {
      submitButton && fireEvent.click(submitButton);
    });
    const spans = elem.querySelectorAll('span');
    expect(spans.length).toBe(1);
    expect(spans[0].textContent).toBe('La cantidad de jugos es requerida.');
  });

  it('should submit', async () => {
    const elem = componentWrapper.container;

    const cantidad_jugos = elem.querySelector('input[name="cantidad_jugos"]');
    const submitButton = elem.querySelector('button[type="submit"]');

    await wait(() => {
      cantidad_jugos &&
        fireEvent.change(cantidad_jugos, setTextEvent('cantidad_jugos', '5'));
    });

    await wait(() => {
      submitButton && fireEvent.click(submitButton);
    });

    const formSubmitted = componentProps.onSubmit.firstCall.args[0];

    expect(formSubmitted.cantidad_jugos).toBe(5);
  });
});
