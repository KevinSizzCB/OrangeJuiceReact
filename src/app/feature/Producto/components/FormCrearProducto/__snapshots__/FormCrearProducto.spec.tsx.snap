// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCrearProducto test should match snapshot 1`] = `
<div>
  <form>
    <h2>
      Form test
    </h2>
    <input
      class="sc-AxirZ eAukyj"
      name="cantidad_jugos"
      placeholder="Cantidad de jugos"
      value=""
    />
    <button
      class="sc-AxjAm kgxTKy"
      type="submit"
    >
      Crear reserva
    </button>
  </form>
</div>
`;
